### PR TITLE
Update Python data sampling algorithm

### PR DESCRIPTION
**Pull request by rohdesamuel**
Previously sampled every 30s, now samples every 0.5s for the first 30s. After which it samples at every 30s again.
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by rohdesamuel** [comment]
R: @udim

File: sdks/python/apache_beam/runners/worker/data_sampler.py [hunk 1]
```diff
@@ -49,11 +49,19 @@
...
+    self._target_timeout_secs = timeout_secs
+    self._timeout_secs = min(timeout_secs, 0.5) if timeout_secs > 0 else 0
     self._timer = Timer(self._timeout_secs, self.sample)
     self._sampler = sampler
+    self._sample_duration_secs = 0
 
   def reset(self) -> None:
+    # For the first 30 seconds, sample every 0.5 seconds. After that, sample at
+    # the normal rate.
+    if self._sample_duration_secs >= 30:
```
Comment (udim) [inline sdks/python/apache_beam/runners/worker/data_sampler.py:61]: IIUC:
```suggestion
    if self._sample_duration_secs >= self._target_timeout_secs:
```

Comment (rohdesamuel) [reply sdks/python/apache_beam/runners/worker/data_sampler.py:61]: Not necessarily, though it seems similar. The logic is there to increase resolution at the beginning to more quickly bring samples to the user. The target_timeout_secs is there only to remember what the original timeout is to set later on.

Comment (udim) [reply sdks/python/apache_beam/runners/worker/data_sampler.py:61]: got it

File: sdks/python/apache_beam/runners/worker/data_sampler.py [hunk 2]
```diff
@@ -49,11 +49,19 @@
...
+      self._timeout_secs = self._target_timeout_secs
+    self._sample_duration_secs += self._timeout_secs
```
Comment (udim) [inline sdks/python/apache_beam/runners/worker/data_sampler.py:63]: From mypy:
`apache_beam/runners/worker/data_sampler.py:63: error: Incompatible types in assignment (expression has type "float", variable has type "int")  [assignment]`

**Review by udim — CHANGES_REQUESTED** [review]

File: sdks/python/apache_beam/runners/worker/data_sampler.py (hunk 2 above)
Comment (rohdesamuel) [reply sdks/python/apache_beam/runners/worker/data_sampler.py:63]: done!

**Review by rohdesamuel — COMMENTED** [review]
done, thanks!

**Review by udim — APPROVED** [review]

**Pull request merged**